include/SFML/Graphics/CoordinateType.hpp

```diff
@@ -0,0 +1,46 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2021 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_COORDINATETYPE_HPP
+#define SFML_COORDINATETYPE_HPP
+
+namespace sf
+{
+
+////////////////////////////////////////////////////////////
+/// \ingroup graphics
+/// \brief Types of texture coordinates that can be used for rendering
+///
+/// \see sf::Texture::bind
+///
+////////////////////////////////////////////////////////////
+enum CoordinateType
+{
+    Normalized, //!< Texture coordinates in range [0 .. 1]
+    Pixels      //!< Texture coordinates in range [0 .. size]
+};
```

Comment:
This spills the constants to the `sf` namespace, i.e. `sf::Normalized` and `sf::Pixels`. Is that what we want (like with [`PrimitiveType`](https://www.sfml-dev.org/documentation/2.5.1/group__graphics.php#ga5ee56ac1339984909610713096283b1b))?

Reply:
The right thing would be to use C++11 `enum class`, but I think that's not an option for SFML. To prevent pollution, we could do something like
```c++
namespace Coordinate
{
    enum Type
    {
        Normalized,
        Pixels
    };
}
```

That will still allow implicit conversions, which `enum class` wouldn't allow, but it's less of an issue. I have no objection going either way.

Reply:
Or, with more boilerplate ([kind of like this](https://musingstudio.com/2014/05/20/how-to-emulate-c11s-enum-class-in-c03/)):
```c++
class CoordinateType
{
public:

    static const CoordinateType Normalized;
    static const CoordinateType Pixels;

    CoordinateType() {}
    explicit CoordinateType(int theValue) : value(theValue) {}
    operator int() const { return value; }

private:
 
    int value;
};
```

But this may start to become a bit obscure.

Reply:
I wouldn't overdo it, both the simple `enum` and the `Coordinate` namespace are good for me. Since this will likely become a C++11 `enum class` in SFML 3, we can also keep things as is for now. I was more asking to make sure everyone is OK with the symbols being in `sf` namespace.